modules/bash-commons/src/array.sh

```diff
@@ -81,4 +81,45 @@ function array_prepend {
 
   updated_ary=( "${ary[@]/#/$prefix}" )
   echo ${updated_ary[*]}
-}
\ No newline at end of file
+}
+
+# Applies a functional reduce to an array of arguments. Supports mathmatical expressions only.
```

Comment:
Sorry, got pulled away for a while to work on other things.

Coming back to this, a sanity check: how useful is a reducer in a bash script that only works on mathematical expressions? Would it be possible to make it work on all types of bash code using `eval`, so long as that code writes its output to `stdout`?